### PR TITLE
Fix #753: add any type, unsearchable free-form json data

### LINKED ISSUE #753: Add 'any' type

### PR DESCRIPTION
**Pull request by bserdar**

### CODE REVIEW

File: util/src/test/java/com/redhat/lightblue/util/JsonUtilsTest.java [hunk 1]
```diff
@@ -120,4 +125,29 @@ public void testJsonWithInvalidJSONInput() throws Exception {
...
+        Map obj=(Map)JsonUtils.fromJson(node);
+        Assert.assertTrue(obj.get("object1") instanceof Map);
+        Assert.assertTrue(obj.get("object2") instanceof Map);
+        Assert.assertEquals("value3",((Map)obj.get("object2")).get("simple3"));
+        Assert.assertEquals(2,((List) ((Map)obj.get("object1")).get("array1")).size());
+        Assert.assertEquals("value1",((Map)((List)((Map)obj.get("object1")).get("array1")).get(0)).get("simple1"));
+    }
+    
+    @Test
+    public void testToson() throws Exception {
```
Comment (dcrissman) [inline util/src/test/java/com/redhat/lightblue/util/JsonUtilsTest.java]: nitpick: typo

File: util/src/main/java/com/redhat/lightblue/util/JsonUtils.java [hunk 2]
```diff
@@ -109,6 +124,126 @@ public static JsonNode json(Reader reader, boolean systemPropertySubstitution) t
...
+
+    private static JsonNode toJson(Collection obj) {
+        ArrayNode node=JsonNodeFactory.instance.arrayNode();
+        for(Object x:obj) {
+            node.add(toJson(x));
+        }
+        return node;
+    }
+
+    /**
```
Comment (dcrissman) [inline util/src/main/java/com/redhat/lightblue/util/JsonUtils.java]: What is the reasoning for putting this logic in JsonUtils and not as a part of AnyType? Maybe the same question put differently, is there a re-use case for this?

**Comment by dcrissman** [comment]
What is the use case for AnyType?

**Comment by bserdar** [comment]
Same logic is duplicated in two other places: one in mongo translator, and
one in metadata properties. I am hoping to remove them eventually.

**Pull request merged**